COLLABORATOR_GUIDE.md

```diff
@@ -0,0 +1,119 @@
+
+If you are unsure about the modification and are not prepared to take full responsibility for the change, defer to another Collaborator.
+
+We recommend collaborators follow the guidelines on the [Contributing Guide](./CONTRIBUTING.md) for reviewing and merging Pull Requests.
+
+### Involving the Web Infrastructure Team
+
+Collaborators may opt to elevate pull requests or issues to the group for discussion by mentioning @nodejs/web-infra. This should be done where a pull request:
+
+- Has a significant impact on the codebase
+- Is inherently controlversial; or
+- Has failed to reach a consensus amongst the Collaborators who are actively participating in the discussion.
+
+The Web Infrastructure should be the final arbiter where needed.
+
+## Technologies Used in the Worker
+
+The Node.js Release Worker is built upon [Cloudflare Workers](https://developers.cloudflare.com/workers/) and [Cloudflare R2](https://developers.cloudflare.com/r2/).
+
+The Worker also uses several other Open Source libraries (not limited to) listed below:
+
+- [AWS S3 client](https://www.npmjs.com/package/@aws-sdk/client-s3) is used for interacting with R2's S3 entrypoint.
+- [Handlebars.js](https://www.npmjs.com/package/handlebars) is used for rendering templated pages.
+- [Sentry](https://sentry.io/about) is used for error reporting.
+- [Zod](https://zod.dev/?id=introduction) is used for error reporting.
+
+## Code Editing
+
+### Structure of this Repository
+
+- `src/` - Worker
+- `scripts/` - Multi-purpose scripts
+- `tests/` - Tests for the worker
+
+## Testing
+
+Each new feature or bug fix should be accompanied by a unit or E2E test (when valuable). We use Node's test runner and Miniflare for our E2E tests.
+
+### Unit Testing
+
+Unit Tests are fundamental to ensure that code changes do not disrupt the functionalities of the Worker:
+
+- Unit Tests should be written as `.test.ts` files in the [tests/unit/](./tests/unit/) directory.
+- They should cover utilities as well as any component that can be broken down and individually tested.
+- We utilize Node's [Test Runner](https://nodejs.org/api/test.html) and [Assert](https://nodejs.org/api/assert.html) APIs.
+- External services used in the worker should be mocked with Undici. If the service cannot be mocked with Undici, it should be mocked with a HTTP server via Node's `createServer` API.
+
+### End-to-End Testing
+
+E2E Tests are fundamental to ensure that requests made to the worker behave as expected:
+
+- E2E Tests should be written as `.test.ts` files in the [tests/e2e/](./tests/e2e/) directory.
+- They should cover the various contexts of a request that could be sent to the worker from an external client.
+- We utilize Node's [Test Runner](https://nodejs.org/api/test.html) and [Assert](https://nodejs.org/api/assert.html) APIs.
+- A local version of the Worker is ran with Miniflare.
+- Like Unit Tests, any external services should be mocked for these tests as well.
+
+## Remarks on Structure and Background
```

Comment:
I'd maybe add a bit more of details on maybe specifics, but this can be iterate over time